packages/integration-sdk-entity-validator/src/validator.ts

```diff
@@ -0,0 +1,158 @@
+import Ajv, { AnySchema } from 'ajv';
+import addFormats from 'ajv-formats';
+import { addJ1Formats } from './j1Formats';
+import {
+  ajvErrorToEntityValidationError,
+  unknownPropertySymbol,
+  isEntityValidationError,
+  EntityValidationError,
+} from './entityValidationError';
+import { assertEntity } from './assertEntity';
+
+type ValidateEntityResult = {
+  isValid: boolean;
+  errors: EntityValidationError[] | null;
+  skippedSchemas: SkippedSchema[] | null;
+};
+
+type SkippedSchema = {
+  schemaId: string;
+  reason: 'not-found' | 'type-already-validated';
+  type: 'type' | 'class';
+};
+
+type ValidateEntityOptions = {
+  forceClassValidationWithValidatedType?: boolean;
+};
+
+const convertUnknownErrorToEntityValidationError = (
+  error: unknown,
+): EntityValidationError => {
+  if (isEntityValidationError(error)) {
+    return error;
+  }
+
+  if (error instanceof Error) {
+    return {
+      schemaId: 'unknown',
+      message: error.message,
+      property: unknownPropertySymbol,
+      validation: 'unknown',
+    };
+  }
+
+  return {
+    schemaId: 'unknown',
+    message: 'Unknown error',
+    property: unknownPropertySymbol,
+    validation: 'unknown',
+  };
+};
+
+export class EntityValidator {
+  private ajvInstance: Ajv;
+
+  constructor({ schemas }: { schemas?: AnySchema[] }) {
+    this.ajvInstance = addJ1Formats(
+      addFormats(
+        new Ajv({
+          strictSchema: false,
+          allErrors: true,
+        }),
+      ),
+    );
+
+    if (schemas) {
+      this.addSchemas(schemas);
+    }
+  }
+
+  addSchemas(schema: AnySchema | AnySchema[]) {
+    this.ajvInstance.addSchema(schema);
+    return this;
+  }
+
+  validateEntity(
+    entity: object,
```

Comment:
How about we keep all the runtime checking you have, but change the type of the `entity` param to be `{_class: string; _type: string}` to have typescript help the consumer do the right thing?

Reply:
I'm not sure the type is useful. It can make people force the type to use the functions, like:

```typescript
validator.validateEntity(myEntityObject as unknown as {_class: string; _type: string})
```

Reply:
Doesn't that still indicate to the consumer that they're doing the wrong thing though? The type is doing its job still by guiding the consumer to not mess up unintentionally.

If they do want to over-rule the type error, they deserve all the evil things that will happen to them. In this life and the next.